mozilla_version/gecko.py

```diff
@@ -123,7 +122,7 @@ class GeckoVersion(BaseVersion):
             |b(?P<beta_number>\d+)
             |(?P<is_esr>esr)
         )?
-        -?(build(?P<build_number>\d+))?$""", re.VERBOSE)
```

Comment:
Question: I'm curious, why do we remove the `^` and the `$`?

Reply:
Lated edit: I presume we'd always want GeckoVersion to run along with `is_within_bigger_string=True` wouldn't we? In that case we wouldn't be having `^$` at all, right?